extensions/AbuseFilter/special/SpecialAbuseLog.php

```diff
@@ -185,36 +186,42 @@ function showList() {
 		// Generate conditions list.
 		$conds = array();
 
+		$skipQuery = false;
+
 		if ( $this->mSearchUser ) {
 			$user = User::newFromName( $this->mSearchUser );
 
 			if ( !$user ) {
 				$conds[] = 'afl_ip=afl_user_text';
 				$conds['afl_user'] = 0;
 				$conds['afl_user_text'] = $this->mSearchUser;
-			} else {
+			} else if (!empty($user->getId())) {
 				$conds['afl_user'] = $user->getId();
-				$conds['afl_user_text'] = $user->getName();
+			} else {
+				$skipQuery = true;
 			}
 		}
 
-		if ( $this->mSearchFilter ) {
-			// if the filter is hidden, users who can't view private filters should not be able to find log entries generated by it
-			if ( !AbuseFilter::filterHidden( $this->mSearchFilter ) || AbuseFilterView::canViewPrivate() ) {
-				$conds['afl_filter'] = $this->mSearchFilter;
+		if (!$skipQuery) {
+
+			if( $this->mSearchFilter ) {
+				// if the filter is hidden, users who can't view private filters should not be able to find log entries generated by it
+				if( !AbuseFilter::filterHidden( $this->mSearchFilter ) || AbuseFilterView::canViewPrivate() ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1066)